src/Uno.UI/UI/Xaml/UIElement.skia.cs

```diff
@@ -140,6 +129,9 @@ internal void AddChild(UIElement child, int? index = null)
 				Visual.Children.InsertAtTop(child.Visual);
 			}
 
+			var enterParams = new EnterParams(IsActiveInVisualTree);
+			ChildEnter(child, enterParams);
```

Comment:
My naïve implementation would have been:

```suggestion
			if(IsActiveInVisualTree)
			{
				ChildEnter(child, new());
			}
```

But I saw that you are actually doing that check in the `ChildEnter`. Is there a consideration I don't see why you choose that approach? -like making sure the check is perform on all platforms, removing the need to check is also in the `UIElement.wasm.cs` file for instance-.

Reply:
@dr1rrb In WinUI, `ChildEnter` should have a lot more complex logic than just `if (@params.IsLive) child.Enter(..)`

![image](https://github.com/unoplatform/uno/assets/31348972/30ef6515-709c-47ba-b78e-d24d7c10e4da)

So keeping it as it's currently will be helpful when we'll work on more alignments for WinUI, e.g, the name registration check and the keyboard accelerator check.